worker.go

```diff
@@ -29,16 +27,20 @@ func NewWorker(workerPool chan chan Job, errCh chan error, resultCh chan Result)
 func (w Worker) Start() {
 	go func() {
 		for {
-			// register the current worker into the worker queue.
-			w.workerPool <- w.jobCh
-
 			select {
-			case job := <-w.jobCh:
-				result, err := job.Execute()
-				if err != nil {
-					w.errCh <- err
+			case job, more := <-w.jobCh:
```

Comment:
nice, I didn't know about the second return arg

Reply:
> nice, I didn't know about the second return arg

yea, pretty handy... it's false if there are no more items and the channel is closed